### ISSUE TITLE
pyodide-build fails with KeyError: 'SIDE_MODULE_CFLAGS'

### ISSUE THREAD

**Issue by eulertour** [pyodide/pyodide#1724]
Installing pyodide-build from the docker image with `python setup.py install --user` and running `~/.local/bin/pyodide-build` results in the following stack trace:
```
Traceback (most recent call last):
  File "/src/.docker_home/.local/bin/pyodide-build", line 33, in <module>
    sys.exit(load_entry_point('pyodide-build==0.18.0.dev0', 'console_scripts', 'pyodide-build')())
  File "/src/.docker_home/.local/lib/python3.9/site-packages/pyodide_build-0.18.0.dev0-py3.9.egg/pyodide_build/__main__.py", line 39, in main
    main_parser = make_parser()
  File "/src/.docker_home/.local/lib/python3.9/site-packages/pyodide_build-0.18.0.dev0-py3.9.egg/pyodide_build/__main__.py", line 33, in make_parser
    parser = module.make_parser(subparsers.add_parser(command_name))  # type: ignore
  File "/src/.docker_home/.local/lib/python3.9/site-packages/pyodide_build-0.18.0.dev0-py3.9.egg/pyodide_build/buildpkg.py", line 313, in make_parser
    default=common.get_make_flag("SIDE_MODULE_CFLAGS"),
  File "/src/.docker_home/.local/lib/python3.9/site-packages/pyodide_build-0.18.0.dev0-py3.9.egg/pyodide_build/common.py", line 44, in get_make_flag
    return get_make_environment_vars()[name]
KeyError: 'SIDE_MODULE_CFLAGS'
```

**Comment by hoodmane** [comment]
This is really annoying. The issue is that it expects the environment variables defined in `Makefile.envs`. I don't know how to `source` a `Makefile.envs` file though, I have been modifying the makefile like
```
.PHONY: do_my_thing
do_my_thing:
	python setup.py install --user
```
and then running `make do_my_thing`. I admit that this is a completely horrendous non-solution. Actually most of the functionality in `pyodide-build` won't work without `include Makefile.envs` which only the makefile knows how to do. So in practice it is difficult to invoke it at all except from `make`.

**Comment by hoodmane** [comment]
To run `pyodide-build mkpkg` what I have been doing is commenting out lines 20 -- 23 in https://github.com/pyodide/pyodide/blob/main/pyodide-build/pyodide_build/__main__.py

**Comment by hoodmane** [comment]
I guess one solution would be to move everything defined in `Makefile.envs` into `pyodide_env.sh`, then `source pyodide_env.sh` before touching `pyodide-build`.

**Comment by hoodmane** [comment]
Another thing that would help would be if the `argparse` subparsers used `None` as the default and only looked up environment variables when actually invoked.

**Comment by eulertour** [comment]
I'm actually trying to build a package, so commenting buildpkg won't work 😕

**Comment by hoodmane** [comment]
You can use the makefile hack I gave above. Generally the way I build packages is by saying `PYODIDE_PACKAGES="my_package" make -C packages`.

**Comment by eulertour** [comment]
I tried using the makefile hack to build a package and got the stack trace at https://pastebin.com/1QYtzhvY. Using `PYODIDE_PACKAGES="my_package" make -C packages` worked.

So running `mkpkg` to generate the `meta.yaml` file requires running the tool after commenting those lines from `__main__.py` and building the `.data` and `.js` files requires running the `make` command you described.

Not the easiest workflow in the world but at least I found a way to do it.

**Comment by hoodmane** [comment]
> Not the easiest workflow in the world

Yes it's really bad.

**Comment by rth** [comment]
We clearly need to improve this situation,
 - IMO `pip install pyodide-build && pyodide-build mkpkg` should work separately from the Pyodide repo. We probably need to make some imports optional there.
 - for building packages yes, currently we need the Makefile to make sure emsdk is installed and also that CPython is built before building packages. Once we upstream all patches to emscripten we should be able to migrate to a situation where one can,
```
conda install emscripten (and other dependencies)
pyodide-build activate # something like, which would call pyodide_env.sh
pyodide-build build <package-name>
```
that would require lots of work on the build system though (related https://github.com/pyodide/pyodide/issues/795). I agree that moving env variables definitions to `pyodide_env.sh` would be a start.

**Comment by genderev** [comment]
>     Once we upstream all patches to emscripten 

How does one build packages in the meantime?

**Comment by rth** [comment]
You add a `meta.yaml` of you package to `packages/<package-name>/`, then run,
```
PYODIDE_PACKAGES="<package name>" make
```
preferably from the Docker image (using the `./run_docker` script on Linux).

**Comment by sylvain-ri** [comment]
I suppressed the error by adding this line in the Makefile.envs, line 73:
`export SIDE_MODULE_CFLAG= $(CFLAGS_BASE)`

**Comment by sylvain-ri** [comment]
Any updates ?
When I comment the lines 20-23, nothing outputs with `pyodide-build mkpkg tqdm`.
Is there a way to build a package without uploading it to pypi ?
Maybe I didn't understood the [Creating a Pyodide package](https://pyodide.org/en/stable/usage/api/pyodide-build-cli.html#pyodide-mkpkg)

**Comment by hoodmane** [comment]
You can make the `meta.yml` file by hand like this:
```yml
package:
  name: package-name
  version: <some version>
requirements:
  run:
  	<runtime dependencies here>
source:
  url: <A URL to a zip archive>
test:
  imports:
  	package_name # the name you would actually import
```
If you have a github you can use `https://github.com/git-user-name/git-repo/archive/refs/heads/master.zip` as the username (copy the url from the "download zip" button).

**Comment by sylvain-ri** [comment]
Thank you both. The micropip works pretty well, already tried it.

I would want to use [SpaCy](https://github.com/explosion/spaCy), with has Cython code and dependencies with other libraries, so micropip doesn't work. Is it something too ambitious at this point ?

**Comment by thadk** [comment]
Seems to be fixed with latest 0.19dev https://github.com/pyodide/pyodide/discussions/2007

**Comment by michaelgold** [comment]
Still an issue for me running under docker with wsl.

I was able modify the workaround mentioned above to successfully get pyodide-build to generate a meta.yaml file

in `__main__.py` comment out these lines (now lines 23-25)
``` python
        # ("buildpkg", buildpkg),
        # ("buildall", buildall),
        # ("pywasmcross", pywasmcross),
```

Then run
``` bash
pip install --user pyodide-build/
```
after that, run `pyodide-build` from this path:
``` bash
/src/.docker_home/.local/bin/pyodide-build
```

**Comment by hoodmane** [comment]
I will look into this, maybe we can fix it for the v0.19 release. I am hoping to make significant improvements to `pyodide-build` for v0.20.

**Comment by hoodmane** [comment]
@ryanking13 I assume this is resolved?

**Comment by ryanking13** [comment]
Yes, this was fixed by #2351
